Core/dsp/ringbuffer.h

```diff
@@ -68,17 +71,9 @@ class ringbufferbase {
 
     void WaitUntilNotEmpty()
     {
-        // if not empty
-        for (int i = 0; i < spin_count; i++)
-        {
-            if (read_index != write_index)
-                return;
-        }
-
+        std::unique_lock<std::mutex> lk(mutex);
```

Comment:
not sure why you change such way. do you see the bug in the original code? if so, please submit a separate PR for this change.

Reply:
The original code did not work at all...
There is shared resource accessed from different threads, so the mutex should be used.